gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -141,42 +172,54 @@ public EntityFetcherResponse visit(DataFetcherNode dataFetcherNode) {
 
     EntitiesRequest request = requestBuilder.build();
     IEntityFetcher entityFetcher = queryHandlerRegistry.getEntityFetcher(source);
-    return entityFetcher.getEntities(context, request);
+
+    EntitiesRequest totalEntitiesRequest =
```

Comment:
Shouldn't the total be computed conditionally based on whether it is requested or not?

Reply:
Ingore. I think we always send the total.

Reply:
Yes. But, we can be smart. We can have a boolean flag on the entities request which would be set when GQL asks for the total, else if the GQL query does not ask for `total`, we won't fetch the total.

For cases, like entity id filter, UI doesn't need total at all, so it doesn't even fetch the total via GQL

Reply:
Yeah... lets do that

Reply:
let me do that as a part of follow up PR, because that would anyways need a change in GQL layer as well